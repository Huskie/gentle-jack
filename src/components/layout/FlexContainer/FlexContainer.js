import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'react-emotion';

const getHorizontalAlignment = (val) => {
  let value = '';
  switch (val) {
    case 'center':
      value = 'center';
      break;
    case 'left':
      value = 'flex-start';
      break;
    case 'right':
      value = 'flex-end';
      break;
    case 'space-between':
      value = 'space-between';
      break;
    default:
      value = 'flex-start';
      break;
  }
  return value;
};

const StyledWrapper = Styled('div')`
  display: flex;
  justify-content: ${props => getHorizontalAlignment(props.horizontalAlignment)};
`;

const FlexContainer = ({ children, horizontalAlignment, verticalAlignment }) => (
  <StyledWrapper
    horizontalAlignment={horizontalAlignment}
    verticalAlignment={verticalAlignment}
  >
    {children}
  </StyledWrapper>
);

FlexContainer.propTypes = {
  children: PropTypes.node.isRequired,
  horizontalAlignment: PropTypes.oneOf(['center', 'left', 'right', 'space-between']),
  verticalAlignment: PropTypes.oneOf(['bottom', 'center', 'top']),
};

FlexContainer.defaultProps = {
  horizontalAlignment: 'left',
  verticalAlignment: 'top',
};

export { FlexContainer };
