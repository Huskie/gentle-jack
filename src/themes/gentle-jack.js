const theme = {
  borderRadius: '4px',
  colours: {
    alpha: '#2742b8',
    beta: '#222',
    gamma: '#06f',
    delta: '#fff',
    epsilon: '#1b6cff',
    zeta: '#ffd561',
    eta: '#4048ef',
  },
  gradients: {
    alpha: '#4048ef, #2742b8',
  },
  spacing: {
    componentMargin: '16px',
  },
  sizing: {
    maxWidth: '1080px',
  },
};

export { theme };
