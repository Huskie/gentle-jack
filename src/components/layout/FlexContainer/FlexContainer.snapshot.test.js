import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'emotion-theming';

import { FlexContainer } from './FlexContainer';
import { theme } from '../../../themes/gentle-jack';

describe('FlexContainer snapshots:', () => {
  it('component renders with defaults', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <FlexContainer>
          <p>A test child.</p>
          <span>Another test child.</span>
        </FlexContainer>
      </ThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('component renders with all props set', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <FlexContainer horiztonalAlignment="end" verticalAlignment="bottom">
          <p>A test child.</p>
          <span>Another test child.</span>
        </FlexContainer>
      </ThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
