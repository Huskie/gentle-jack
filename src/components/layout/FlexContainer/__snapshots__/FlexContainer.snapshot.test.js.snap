// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexContainer snapshots: component renders with all props set 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

<div
  className="emotion-0"
>
  <p>
    A test child.
  </p>
  <span>
    Another test child.
  </span>
</div>
`;

exports[`FlexContainer snapshots: component renders with defaults 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

<div
  className="emotion-0"
>
  <p>
    A test child.
  </p>
  <span>
    Another test child.
  </span>
</div>
`;
